File: CPI Silver Class/Week 3: Two Pointers/Homework/Problem: Books/Main.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Kattio io = new Kattio();

        int n = io.nextInt();
        int t = io.nextInt();

        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = io.nextInt();
        }

        int result = 0;
        for (int i = 0, j = -1; i < n; i++) {
            while (j + 1 < n && a[j + 1] <= t) {
                t -= a[++j];
            }
            result = Math.max(result, j - i + 1);
            if (i - 1 == j) {
                j++;
            } else {
                t += a[i];
            }
        }

        io.println(result);

        io.close();
    }
}

class Kattio extends PrintWriter {
    private BufferedReader r;
    private StringTokenizer st;

    // standard input
    public Kattio() {
        this(System.in, System.out);
    }

    public Kattio(InputStream i, OutputStream o) {
        super(o);
        r = new BufferedReader(new InputStreamReader(i));
    }

    // USACO-style file input
    public Kattio(String problemName) throws IOException {
        super(new FileWriter(problemName + ".out"));
        r = new BufferedReader(new FileReader(problemName + ".in"));
    }

    // read next line
    public String readLine() {
        try {
            return r.readLine();
        } catch (IOException e) {
        }
        return null;
    }

    // returns null if no more input
    public String next() {
        try {
            while (st == null || !st.hasMoreTokens())
                st = new StringTokenizer(r.readLine());
            return st.nextToken();
        } catch (Exception e) {
        }
        return null;
    }

    public int nextInt() {
        return Integer.parseInt(next());
    }

    public double nextDouble() {
        return Double.parseDouble(next());
    }

    public long nextLong() {
        return Long.parseLong(next());
    }
}

File: CPI Silver Class/Week 3: Two Pointers/Homework/Problem: Books/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, t;
    cin >> n >> t;

    int a[n];
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    int result = 0;
    for (int i = 0, j = -1; i < n; i++) {
        while (j + 1 < n && a[j + 1] <= t) {
            t -= a[++j];
        }
        result = max(result, j - i + 1);
        if (i - 1 == j) {
            j++;
        } else {
            t += a[i];
        }
    }

    cout << result << endl;

    return 0;
}
